### PR TITLE
modelling work for review 

### PR DESCRIPTION
**Pull request by abdullateefogundipe**

### CODE REVIEW

**Comment by j-unspeakable** [comment]
Amazing job boss!

Creating a branch for this would have been better so others can access the branch locally before merging to the main branch. If you can, please create a branch, push the branch and then open a new pull request. You did a lot of work so the diffs are hard to follow on GitHub.

**Comment by abdullateefogundipe** [comment]
Thanks. I actually thought I created. I was guided to the process using git bash during the session yesterday. While trying to push to this repo, I was asked to go through the authentication process which kept failing and the reason is pasted below

"Logon failed, use ctrl+c to cancel basic credential prompt.
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/abdullateefogundipe/Gold-Rates-Prediction.git/'"

So I found the branch created on my github desktop and pushed it directly thinking it will create a maintain the branch as created.

Just trying to give a comprehensive stuff so that others can benefit as I am aware quite a number of people are facing this issue.

Please, guide us to other ways to get this done. Thanks

**Comment by j-unspeakable** [comment]
Yeah. I believe it is an authentication issue which shouldn't be the case since you've been added to the team here. In any case, GitHub uses tokens for authentications now and that's why it gave you the error message "emote: Support for password authentication was removed on August 13, 2021." So you might need to generate a token. You can reach out to the person that created a branch on the repo and ask him if he faced the same challenges. I would have tried it out but I'm still making some changes and even interested in seeing your changes before creating a branch and opening a PR.

> Thanks. I actually thought I created. I was guided to the process using git bash during the session yesterday. While trying to push to this repo, I was asked to go through the authentication process which kept failing and the reason is pasted below
> 
> "Logon failed, use ctrl+c to cancel basic credential prompt. remote: Support for password authentication was removed on August 13, 2021. remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication. fatal: Authentication failed for 'https://github.com/abdullateefogundipe/Gold-Rates-Prediction.git/'"
> 
> So I found the branch created on my github desktop and pushed it directly thinking it will create a maintain the branch as created.
> 
> Just trying to give a comprehensive stuff so that others can benefit as I am aware quite a number of people are facing this issue.
> 
> Please, guide us to other ways to get this done. Thanks

Yeah. I believe it is an authentication issue which shouldn't be the case since you've been added to the team here. In any case, GitHub uses tokens for authentications now and that's why it gave you the error message "remote: Support for password authentication was removed on August 13, 2021." So you will need to generate a token for the authentication. You can reach out to the person that created a branch on the repo and ask him if he faced the same challenges. I would have tried it out but I'm still making some changes and even interested in seeing your changes before creating a branch and opening a PR.

**Comment by abdullateefogundipe** [comment]
Alright. Thanks
